modules/codegen/src/smithy4s/codegen/Renderer.scala

```diff
@@ -671,8 +671,15 @@ private[codegen] class Renderer(compilationUnit: CompilationUnit) { self =>
       case Type.Alias(ns, name, _, _) =>
         line"$name.underlyingSchema".addImport(ns + "." + name)
       case Type.Ref(ns, name) => line"$name.schema".addImport(ns + "." + name)
-      case Type.ExternalType(_, _, providerFqn, underlyingTpe, hint) =>
-        line"${underlyingTpe.schemaRef}.refined(${renderNativeHint(hint)})($providerFqn)"
+      case Type.ExternalType(
+            _,
+            fqn,
+            maybeProviderImport,
+            underlyingTpe,
+            hint
+          ) =>
+        line"${underlyingTpe.schemaRef}.refined(${renderNativeHint(hint)})"
+          .addImport(maybeProviderImport.getOrElse(s"$fqn._"))
```

Comment:
If I don't add the import for the companion object of the `targetType` explicitly, I get a diverging implicit expansion. I think scala expects the implicit to be in the companion object of the constraint by default and doesn't check the target type companion object by default?

Reply:
It might be a matter of annotating some types to get that type ineference to kick in

Reply:
Yeah, Oli fixed it actually. See: https://github.com/disneystreaming/smithy4s/pull/346#discussion_r934938553